### PR TITLE
Added cleanup identifier quotes on xpath queries

### PR DESCRIPTION
**Pull request by m-cattena**
Xpath queries don't support quote escaping so we have to deal with it by wrapping the identifier in single quotes if it has double quotes or viceversa.
If the identifier has both types of quotes we use replace and concat to assemble the proper string.

Also removed the class check from hover events that was picking up unnecessary events

### CODE REVIEW

File: src/recorder/events/handlers/hover-event-handler.js [hunk 1]
```diff
@@ -4,8 +4,7 @@ import ElementHovered from '../element-hovered';
 
 function isHoverable(element) {
   while (element) {
-    if (element.tagName && element.tagName.toLowerCase().includes('nav') ||
-      element.className && typeof element.className === 'string' && element.className.toLowerCase().includes('hover')) {
```
Comment (artgo) [inline src/recorder/events/handlers/hover-event-handler.js:8]: Is this a related change?

File: src/recorder/helpers/query-helper.js [hunk 2]
```diff
@@ -2,26 +2,35 @@
 const QUERYABLE_ATTRS = ['text', 'hint', 'title', 'label', 'aria-label', 'name', 'id', 'data-test-id', 'class',
   'placeholder', 'alternative', 'source'];
 
+function cleanupQuotes(value) {
+  if (value.indexOf("'") === -1) {
+    return '\'' + value + '\'';
```
Comment (artgo) [inline src/recorder/helpers/query-helper.js:7]: testRigor can't process single quotes - only double

File: src/recorder/helpers/query-helper.js [hunk 3]
```diff
@@ -2,26 +2,35 @@
...
+  } else if (value.indexOf('\"') === -1) {
+    return '"' + value + '"';
+  }
+  return "concat('" + value.replace("'", "',\"'\",'") + "')";
```
Comment (artgo) [inline src/recorder/helpers/query-helper.js:11]: Single quotes do not matter, not sure where concat comes from

File: src/recorder/helpers/query-helper.js [hunk 4]
```diff
@@ -2,26 +2,35 @@
 const QUERYABLE_ATTRS = ['text', 'hint', 'title', 'label', 'aria-label', 'name', 'id', 'data-test-id', 'class',
   'placeholder', 'alternative', 'source'];
 
+function cleanupQuotes(value) {
```
Comment (artgo) [inline src/recorder/helpers/query-helper.js:5]: Should we call it `cleanup` and normalize space as well to make sure there are no newlines?

File: src/recorder/events/handlers/hover-event-handler.js (hunk 1 above)
Comment (m-cattena) [reply src/recorder/events/handlers/hover-event-handler.js:8]: Not exactly related to quotes, but this was causing us to record a bunch of unnecessary steps for the same website that we found this quote issue for.

Should I start a separate PR?

File: src/recorder/helpers/query-helper.js (hunk 2 above)
Comment (m-cattena) [reply src/recorder/helpers/query-helper.js:7]: This is not used to clean the identifier that goes to testRigor. It is used for [xPath queries](https://developer.mozilla.org/en-US/docs/Web/API/Document/evaluate) that we do to look for related elements on the page (looking for elements with the same identifier as the event's target, for example)
Since xPath queries don't support escaping, the xPath expressions that we evaluate must deal with quotes in this way

For example, a simplified version of a query that looks for elements that contain `myIdentifier` would be the following string: `//*/body//*[contains("${myIdentifier}")` and we would interpolate myIdentifier there to get the query
If myIdentifier contains a double quote it breaks the query by making the number of double quotes uneven. 
To solve this, we wrap myIdenfifier in simple quotes if doesn't contain any, guaranteeing the number is even.
If it doesnt have double quotes, we wrap it in double quotes instead of single to achieve the same result
If the identifier has both types of quotes we use replace and [concat](https://developer.mozilla.org/en-US/docs/Web/XPath/Functions/concat) to assemble the proper string with an even number of quotes.

File: src/recorder/helpers/query-helper.js (hunk 3 above)
Comment (m-cattena) [reply src/recorder/helpers/query-helper.js:11]: they do, see above

File: src/recorder/helpers/query-helper.js (hunk 4 above)
Comment (m-cattena) [reply src/recorder/helpers/query-helper.js:5]: cleanup, normalization and trimming are performed on the service side

**Review by enzotestrigor — APPROVED** [review]
LGTM

**Review by parunruaivo — APPROVED** [review]
LGTM

File: src/recorder/events/handlers/hover-event-handler.js (hunk 1 above)
Comment (artgo) [reply src/recorder/events/handlers/hover-event-handler.js:8]: That's fine I guess

**Review by artgo — APPROVED** [review]
LGTM
